File: CoreJava/game-tic-tac-toe/src/com/techlab/game/Game.java
package com.techlab.game;

import java.util.Scanner;

public class Game {

	Board board;
	private static Mark gameBoard[][];
	Player turn;
	private int position;
	ResultAnalyser analyse;
	int noOfMoves = 0;

	Scanner sc = new Scanner(System.in);

	public Game(Board board) {

		this.board = board;
		gameBoard = this.board.getBoard();
		analyse = new ResultAnalyser(board);

		setBoard();

	}

	public void startGame(Player p1, Player p2) {

		turn = p1;
		
		while (noOfMoves < (Board.getBoardSize() * Board.getBoardSize())) {

			play(turn);

			if (analyse.checkWin()) {
				System.out.println(turn.getPlayerName() + " " + Result.WIN);
				showBoard();
				break;
			} else {
				System.out.println(Result.IN_PROGRESS);
			}

			if (turn.equals(p1))
				turn = p2;
			else
				turn = p1;

			showBoard();
			noOfMoves++;
		}

		if (!analyse.checkWin()) {
			System.out.println(Result.DRAW);
		}

	}

	private void play(Player turn) {

		System.out.println(turn.getPlayerName() + "'s turn :");

		do {
			System.out.println("Enter Position");
			position = sc.nextInt();
		} while (!Cell.isPositionFree(position));

		Board.putMark(position, turn.getMark());

	}

	private void setBoard() {
		for (int i = 0; i < Board.getBoardSize(); i++) {
			for (int j = 0; j < Board.getBoardSize(); j++) {
				gameBoard[i][j] = Mark.BLANK;
			}
		}
		showBoard();
	}

	private static void showBoard() {

		System.out.println("\n\n");

		for (int i = 0; i < Board.getBoardSize(); i++) {
			for (int j = 0; j < Board.getBoardSize(); j++) {
				System.out.print(gameBoard[i][j]);
				if (j < Board.getBoardSize() - 1)
					System.out.print(" | ");
			}
			if (i < Board.getBoardSize() - 1)
				System.out.println("\n----------\n");
			else
				System.out.println();
		}

	}

	public static void showRules() {
		System.out.println("\t\t\t  Tic-Tac-Toe\n\n");
		System.out.println("Choose a cell numbered from 1 to " + Board.getBoardSize() * Board.getBoardSize()
				+ " as below and play\n\n");

		int num = 1;

		for (int i = 0; i < Board.getBoardSize(); i++) {
			for (int j = 0; j < Board.getBoardSize(); j++) {
				System.out.print(num);
				num++;
				if (j < Board.getBoardSize() - 1)
					System.out.print(" | ");
			}
			if (i < Board.getBoardSize() - 1)
				System.out.println("\n----------\n");
		}

		System.out.println(
				"\nRules:\nThree marks of same type appear in a row either vertically or horizontally or diagonally,"
						+ "\nthe user of that mark will be declared winner by the system. "
						+ "\n\nIf all the squares of the board are filled and no user is the winner then"
						+ "\nthe game is declared as draw by the system.\n\n");

		return;
	}

}

File: C#Folder/TicTacToeGameSolution/TicTacToeGameLib/Game.cs
using System;
using System.Runtime.CompilerServices;
using System.Windows.Forms;
using TicTacToeFormApp;

namespace TicTacToeGameLib
{
    public class Game : Form
    {
        public static Board board;
        public static Player turn;
        public static ResultAnalyser analyse;
        public static int noOfMoves = 0;
        public static Program currentGame;

        public Game(Board b, Program program)
        {
            board = b;
            analyse = new ResultAnalyser(Cell.cell);
            turn = Program.p2;
            currentGame = program;
            Console.WriteLine("First turn is : " + turn.PlayerName);
        }

        public static Player getTurn()
        {
            if (turn.Equals(Program.p1))
            {
            
                Program.ActiveForm.Text=  "Its " + turn.PlayerName + "'s Turn...";
                return Program.p2;
            }
            else
            {
                Program.ActiveForm.Text = "Its " + turn.PlayerName + "'s Turn...";
                return Program.p1;
            }

        }

        public static TableLayoutPanel showRules(Board b)
        {

            TableLayoutPanel rulesPanel = new TableLayoutPanel();

            TextBox rules = new TextBox();
            rules.SetBounds(250, 200, 100, 100);
            rules.Text = "\nRules:\nThree marks of same type appear in a row\n either vertically or horizontally or diagonally,"
                            + "\nthe user of that mark will be declared\n winner by the system. "
                            + "\n\nIf all the squares of the board are\n filled and no user is the winner then";
            rulesPanel.Controls.Add(rules);

            return rulesPanel;
        }

        public static void analyze()
        {

            if (analyse.checkWin())
            {

                showResult(turn.PlayerName + " " + Result.WIN.ResultMsg());

            }

            if (noOfMoves == Board.BoardSize * Board.BoardSize)
            {
                showResult("" + Result.DRAW.ResultMsg());
            }
        }

        private static void showResult(String result)
        {
            string title = "Result";
            result += "\nDo You Want to Restart..??";
            MessageBoxButtons buttons = MessageBoxButtons.YesNo;
            DialogResult d = MessageBox.Show(result, title, buttons);
            if (d == DialogResult.Yes)
            {
                //currentGame.Controls.Remove(board.GameBoard);
                //new Program();
                Application.Restart();
            }
            else
            {
                currentGame.Close();
            }
        }
    }
}
